datatypes/src/operations/image/to_png.rs

```diff
@@ -16,15 +18,18 @@ where
     fn to_png(&self, width: u32, height: u32, colorizer: &Colorizer) -> Result<Vec<u8>> {
         // TODO: use PNG color palette once it is available
 
-        let (.., raster_y_size, raster_x_size) = self.dimension().as_pattern();
+        let [.., raster_y_size, raster_x_size] =
+            self.grid_dimension.size_as_index().as_index_array();
```

Comment:
why not call `slice()` it says it returns the size of the axis

Reply:
slice is a slice you will loose the axis count if you do this